src/org/sosy_lab/common/NativeLibraries.java

```diff
@@ -105,55 +110,53 @@ public static Architecture guessVmArchitecture() {
         return currentArch;
       }
 
-      @Var String prop = System.getProperty("os.arch.data.model");
-      if (isNullOrEmpty(prop)) {
-        prop = System.getProperty("sun.arch.data.model");
+      currentArch = downgradeArchIfRequired(getArchFromSystemProperty());
+
+      return currentArch;
+    }
+
+    private static Architecture getArchFromSystemProperty() {
```

Comment:
```suggestion
    private static Architecture guessOsArchitecture() {
```
This needs to make explicit that the result is the OS architecture, and not the JVM one. While we are at it, there seems to be no reason to not make the naming consistent with the existing method.

Reply:
This method does no longer guess, but simply use "os.arch". The new name seems to be a better match.

Reply:
I will rename it again.